### PR TITLE
settingsnode: enable trusted daemon option

### LINKED ISSUE #1938: Monero GUI: Trusted Daemon?
https://www.reddit.com/r/Monero/comments/aquwbj/monero_gui_trusted_daemon/

### PR DESCRIPTION
**Pull request by mmbyday**
fixes #1938 
tttt

### CODE REVIEW

File: pages/settings/SettingsNode.qml [hunk 1]
```diff
@@ -330,6 +330,16 @@ Rectangle{
...
             }
 
+            MoneroComponents.CheckBox {
+                id: setTrustedDaemonCheckBox
+                checked: persistentSettings.is_trusted_daemon
+                onClicked: {
+                    persistentSettings.is_trusted_daemon = !persistentSettings.is_trusted_daemon
+                    currentWallet.setTrustedDaemon(persistentSettings.is_trusted_daemon)
+                }
+                text: qsTr("Enable as Trusted Daemon") + translationManager.emptyString
```
Comment (xmrdsc) [inline pages/settings/SettingsNode.qml]: `Mark as trusted daemon`

**Comment by xmrdsc** [comment]
The option should probably reset itself when the remote node changes.

File: pages/settings/SettingsNode.qml (hunk 1 above)
Comment (mmbyday) [reply pages/settings/SettingsNode.qml]: I dig it 👍

**Comment by mmbyday** [comment]
> The option should probably reset itself when the remote node changes.

Good idea, now implemented.

A status message will flash the following
Remote node updated. Trusted daemon has been reset. Mark again, if desired.

**Comment by Setzler22** [comment]
This doesn't appear to be working on the Linux GUI client. No matter what I do it still doesn't let me rescan the wallet balance

**Comment by Setzler22** [comment]
Actually, it's even worse. There have been times that it is unchecked and it is acting like a trusted node, and there are times that it is checked and it is not. In other words, the check box on the GUI has no connection to whether the node is actually trusted or not.

This is a serious security issue.

**Pull request merged**